nengo_dl/converter.py

```diff
@@ -123,49 +122,34 @@ def __init__(
         self.swap_activations = swap_activations or {}
         self.scale_firing_rates = scale_firing_rates
         self.synapse = synapse
-        self.layer_map = collections.defaultdict(dict)
         self._layer_converters = {}
 
-        with nengo.Network(label=model.name) as self.net:
-            # add the "trainable" attribute to all objects
-            configure_settings(trainable=None, inference_only=self.inference_only)
-
-            # convert model
-            self.get_converter(model).convert(None)
+        # convert model
+        self.net = self.get_converter(model).convert(None)
+        with self.net:
+            configure_settings(inference_only=self.inference_only)
 
             if isinstance(model, tf.keras.Sequential):
```

Comment:
I don't think it's necessary to explicitly check this, since for functional models `self.get_converter(model).layer` will just return the original model.

Reply:
Yeah good call :+1: